src/mpid/ch4/src/init_comm.c

```diff
@@ -60,18 +60,24 @@ int MPIDI_create_init_comm(MPIR_Comm ** comm)
     goto fn_exit;
 }
 
-void MPIDI_destroy_init_comm(MPIR_Comm ** comm_ptr)
+int MPIDI_destroy_init_comm(MPIR_Comm ** comm_ptr)
 {
+    int mpi_errno = MPI_SUCCESS;
     int in_use;
     MPIR_Comm *comm = NULL;
     if (*comm_ptr != NULL) {
         comm = *comm_ptr;
         MPIDIU_release_lut(MPIDI_COMM(comm, map).irreg.lut.t);
         MPIDIG_destroy_comm(comm);
+        /* wait for any outstanding requests to complete */
+        MPIDIU_PROGRESS_WHILE(MPIR_Object_get_ref(comm) > 1, 0);
```

Comment:
I'm thinking of dropping this patch and just including the additional error checks. It's possible the assertion failure `MPIR_Assertp(in_use == 0);` observed on Aurora was just the result of a previously masked error. @hzhou what do you think?

Reply:
I agree. There are not many nondeterministic things during init. When we destroy init_comm, all operations should be locally complete already, or it is likely a bug. How about skip this for now and confirm on Aurora?